airbyte-integrations/connectors/source-adroll/manifest.yaml

```diff
@@ -0,0 +1,2199 @@
+          $ref: "#/definitions/base_requester"
+          path: /api/v1/pixel/get_segments
+          http_method: GET
+          request_parameters:
+            pixel: "{{ config[\"pixel_eid\"] }}"
+        record_selector:
+          type: RecordSelector
+          extractor:
+            type: DpathExtractor
+            field_path:
+              - results
+        partition_router:
+          type: SubstreamPartitionRouter
+          parent_stream_configs:
+            - type: ParentStreamConfig
+              parent_key: eid
+              partition_field: parent_id
+              stream:
+                $ref: "#/definitions/streams/Active Pixels for an Advertisable"
+      schema_loader:
+        type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/Segments for a Pixel"
+    Campaigns:
+      type: DeclarativeStream
+      name: Campaigns
+      primary_key:
+        - eid
+      retriever:
+        type: SimpleRetriever
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: /reporting/api/v1/query
+          http_method: POST
+          request_body_json:
+            query: >-
+              query CampaignsForAdvertisables($advertisable: String!) {
+              advertisable { byEID(advertisable: $advertisable) { name eid
+              campaigns { eid name type channel status } } } }
+            variables:
+              advertisable: "{{ stream_partition.parent_id }}"
+        record_selector:
+          type: RecordSelector
+          extractor:
+            type: DpathExtractor
+            field_path:
+              - data
+              - advertisable
+              - byEID
+              - campaigns
+        paginator:
+          type: DefaultPaginator
+          page_token_option:
+            type: RequestOption
+            inject_into: request_parameter
+            field_name: page
+          page_size_option:
+            type: RequestOption
+            field_name: per_page
+            inject_into: request_parameter
+          pagination_strategy:
+            type: PageIncrement
+            page_size: 10
+            start_from_page: 1
+            inject_on_first_request: true
+        partition_router:
+          type: SubstreamPartitionRouter
+          parent_stream_configs:
+            - type: ParentStreamConfig
+              parent_key: eid
+              partition_field: parent_id
+              stream:
+                $ref: "#/definitions/streams/Advertisables"
+      schema_loader:
+        type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/Campaigns"
+    AdGroups by Advertisables:
+      type: DeclarativeStream
+      name: AdGroups by Advertisables
+      primary_key:
+        - eid
+      retriever:
+        type: SimpleRetriever
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: /reporting/api/v1/query
+          http_method: POST
+          request_body_json:
+            query: >-
```

Comment:
Injecting Graphql query can be tricky because it won't have pagination working properly.
@natikgadzhi any opinion about this implementation? My thoughts:
* I can be nice to have a first version with some limitations
* if we get a lot of users facing the limitation it's a good signal to redo/promote this connector.
But to follow this path I'd add a big :warning: message in the connector documentation saying:
* Connector is limited to extract max 100 campaigns, if you're project has more ... help improving the connector.